Optimize Dockefile

robbyoconnor opened the pull request:
Move stuff to the top to delay cache invalidation and remove packages which are
already in the base image.

Closes #192

paulcsmith commented:
I needed to update the gems. Could you rebase and force push? Then it should be able to deploy the preview ok

paulcsmith commented on source/guides/logging-and-error-handling.html.md.erb:
````diff
@@ -0,0 +1,121 @@
+# Does some fancy math to display the elapsed span in a human readable format
+elapsed_text(elapsed)
+```
+
````
Very nice writeup! 👍

paulcsmith commented on source/guides/logging-and-error-handling.html.md.erb:
````diff
@@ -0,0 +1,121 @@
+  ErrorReporter.report(context, error)
+  render Errors::ShowPage, status: 500, title: "Oops!"
+end
+```
````
I think that in general using `handle_error `in `Errors::Show` should be preferred over including `Lucky::HttpRespondable`. The reason is that you get a lot more flexibility in the `Errors::Show`. You can control what the body should be, you can change it based on content-type, etc. 

So could you show the example exception class and then show an example `Errors::Show#handle_error` method? Then maybe keep the part you have now below as another example. I think it is still useful, but is usually more useful for libraries. What do you think? Also I think the order is great! We can tweak it later if necessary

jwoertink replied:
Not sure if I fully understood, but I added an extra example at the bottom. Let me know if that works.

jwoertink replied:
Sorry, got a little busy there. Ok, pushed up a change. I'm feeling like this is a little redundant now, but I guess redundancy is ok in the sake of clarity. Let me know if that looks better, or if you have any other suggestions.

paulcsmith commented on source/guides/logging-and-error-handling.html.md.erb:
```diff
@@ -0,0 +1,121 @@
+### log_formatter : Lucky::LogFormatters::Base
+Customize how your log is formatted, and what information it shows. By default this is the `DefaultLogFormatter`
+
+### enabled : Bool
```
Could you add newlines underneath these `###`

jwoertink replied:
Done.

robbyoconnor committed:
Optimize Dockefile

Move stuff to the top to delay cache invalidation and remove packages which are
already in the base image.

jwoertink reviewed (approved):
Sweet. Looks good. Thanks!
